packages/oauth/src/errors.ts

```diff
@@ -25,6 +26,10 @@ export class MissingStateError extends Error implements CodedError {
   public code = ErrorCode.MissingStateError;
 }
 
+export class MissingCodeError extends Error implements CodedError {
```

Comment:
Added a new specific error for case when `code` param is omitted. Previously this was covered under `MissingStateError`, but I felt it made sense to split these failure states up.

Reply:
This is a nice improvement!

Reply:
Can we add a test that checks for this new error? We used to have it but this PR seems to have re-used it for state validation - can we re-add it back?

Reply:
Yep, added!